### ISSUE TITLE
AttributeError: module 'PyNvCodec' has no attribute 'PyNvDecoder'

### ISSUE THREAD

**Issue by sbbug** [NVIDIA/VideoProcessingFramework#37]
hello，i meet a problem "AttributeError: module 'PyNvCodec' has no attribute 'PyNvDecoder'",
can u solve it? thanks

**Comment by rarzumanyan** [comment]
Did you face this when you run of the samples?

**Comment by conraddd** [comment]
@rarzumanyan  Here is the code I copied from [Wiki](https://github.com/NVIDIA/VideoProcessingFramework/wiki/Exporting-video-frame-to-Pytorch-tensor) and I don't know where nv12_smaller and pnvc come from.
```
    nvDec = nvc.PyNvDecoder(encFile, gpuID)
    to_rgb = nvc.PySurfaceConverter(nvDec.Width(), nvDec.Height(), nvc.PixelFormat.NV12, nvc.PixelFormat.RGB, gpuID)
    to_planar = nvc.PySurfaceConverter(nvDec.Width(), nvDec.Height(), nvc.PixelFormat.RGB, nvc.PixelFormat.RGB_PLANAR, gpuID)

    while True:
        # Obtain NV12 decoded surface from decoder;
        rawSurface = nvDec.DecodeSingleSurface()
        if (rawSurface.Empty()):
            break

        # Convert to RGB interleaved;
        rgb_byte = to_rgb.Execute(nv12_smaller)

        # Convert to RGB planar because that's what to_tensor + normalize are doing;
        rgb_planar = to_planar.Execute(rgb_byte)

        # Create torch tensor from it and reshape because
        # pnvc.makefromDevicePtrUint8 creates just a chunk of CUDA memory
        # and then copies data from plane pointer to allocated chunk;
        surfPlane = rgb_planar.PlanePtr()
        surface_tensor = pnvc.makefromDevicePtrUint8(surfPlane.GpuMem(), surfPlane.Width(), surfPlane.Height(), surfPlane.Pitch(), surfPlane.ElemSize())
        surface_tensor.resize_(3, target_h, target_w)

        # This is optional and depends on what you NN expects to take as input
        # Normalize to range desired by NN. Originally it's 
        # transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
        # But we scale that to [0;255] input, so we multiply normalization coefficients
        surface_tensor = surface_tensor.type(dtype=torch.cuda.FloatTensor)
        mean = torch.tensor([123.675, 116.28, 103.53], dtype=torch.float32, device='cuda')
        std = torch.tensor([58.395, 57.12, 65.025], dtype=torch.float32, device='cuda')
```

**Comment by sbbug** [comment]
yes and python version is 3.7,
but when i am cmaking, i meet 
" Found PythonLibs: /home/jt1/miniconda3/lib/libpython3.7.so (found suitable version "3.7.4", minimum required is "3.5") 
-- Found PythonInterp: /usr/bin/python3.6 (found version "3.6.9") 
-- Found PythonLibs: /home/jt1/miniconda3/lib/libpython3.7.so
"
i had assigned python version(3.7), Why 3.6?

**Comment by rarzumanyan** [comment]
I guess that's the reason: you've built the module for python 3.7 but default python interpreter is 3.6.
Can you launch the sample with explicit python version selection? 
Like `python3.7 SampleDecode.py ...`

**Comment by sbbug** [comment]
hello，we had solved the problem:

" Found PythonLibs: /home/jt1/miniconda3/lib/libpython3.7.so (found suitable version "3.7.4", minimum required is "3.5")
-- Found PythonInterp: /usr/bin/python3.6 (found version "3.6.9")
-- Found PythonLibs: /home/jt1/miniconda3/lib/libpython3.7.so

we use the right order： 
cmake ..   -DFFMPEG_DIR:PATH="$PATH_TO_FFMPEG"   -DVIDEO_CODEC_SDK_DIR:PATH="$PATH_TO_SDK"   -DGENERATE_PYTHON_BINDINGS:BOOL="1"   -DPYTHON_LIBRARY=/home/jt1/miniconda3/lib/libpython3.7m.so   -DCMAKE_INSTALL_PREFIX:PATH="$INSTALL_PREFIX" -**DPYTHON_EXECUTABLE=/home/jt1/miniconda3/bin/python**

note : add  DPYTHON_EXECUTABLE=/home/jt1/miniconda3/bin/python

**Comment by sbbug** [comment]
> I guess that's the reason: you've built the module for python 3.7 but default python interpreter is 3.6.
> Can you launch the sample with explicit python version selection?
> Like `python3.7 SampleDecode.py ...`
thanks

**Comment by rarzumanyan** [comment]
hi @sbbug 
Glad it worked for you. Closing the issue.

**Comment by krydm** [comment]
I'm getting the same error with Python 3.8. Just to check I've tried to load it with python3.7, but then it complains about non-found numpy. Instead of fighting with missing numpy I've just commented the import string -- but then got again the same  "AttributeError: module 'PyNvCodec' has no attribute 'PyNvDecoder'", as with python 3.8.
Would appreciate any advises. TIA
Also, I somehow don't see PyNvCodec in python's help('modules') and neither in pip3 freeze output, in case it can be a hint to some problem
